src/components/PortalDatePicker/PortalDatePicker.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react'
+import PortalPager from '../Pager/ProtalPager';
+import './PortalDatePicker.css';
+
+const PortalDatePicker = (props) => {
+    const { type } = props;
+    const { onDateSelection } = props;
+
+    const today= new Date();
+    const [currentDate, setCurrentDate] = useState(today);
+
+    useEffect(() => {
```

Comment:
call onDateSelection in the same place you update the state - no need for useEffect here